bittensor/core/types.py

```diff
@@ -518,3 +519,28 @@ def with_log(
         if self.message:
             getattr(logging, level)(self.message)
         return self
+
+
+@dataclass
+class BlockInfo:
+    """
+    Class that holds information about a blockchain block.
+
+    This class encapsulates all relevant information about a block in the blockchain, including its number, hash,
+    timestamp, and contents.
+
+    Attributes:
+        number: The block number.
+        hash: The corresponding block hash.
+        timestamp: The timestamp of the block (based on the `Timestamp.Now` extrinsic).
+        header: The raw block header returned by the node RPC.
+        extrinsics: The list of decoded extrinsics included in the block.
```

Comment:
```suggestion
        extrinsics: The list of extrinsics included in the block.
```

They're _kind of_ decoded. They're decoded in the sense that they are readable and usable, but not in the sense that they've had `decode()` called on them, which produces a dict rather than a GenericExtrinsic object. Either update the wording here, or pass the decoded extrinsics (e.g. dicts from `extrinsic.decode()` into this attr(